### ISSUE TITLE
openBVE 1.7.1.0 Window Behaviour

### ISSUE THREAD

**Issue by sjain882** [leezer3/OpenBVE#415]
### Description

Hi, I'm a long time user of openBVE for a long time, thanks for making such a great sim.

I always stuck to some v1.4.xx version, it was 100% compatible with all my London Underground stuff, and I have been using it since its release when I upgraded from v1.3.xx something all the way back then. 

A year or two later I tried v1.5 (or one of the newer versions) which bought some changes to controls, the engine, etc, but I didn't like how it worked, everything felt really off and slower, buggier, etc, and it kept crashing despite a clean install so I deleted it and went back to my v1.4.xx folder.

Today I decided to boot up openBVE after not touching it for 3 or so years and everything worked perfect, but the 1992 Stock of the well known London Underground Central Line Demo route was not loading its westinghouseATP.dll plugin at all, and all the security functions were non functional.

After googling about, it turns out this is because the route takes too long to load (I have a relatively old PC) and something times out, and that the plugin itself is a little buggy. I know this because I tried a clean re-install, running streams from technet, ensuring everything was unblocked, excluding antivirus etc.

So I decided I might as well upgrade to the latest openBVE, I installed 1.7.10 to a new folder and it is working great, infact I was really pleasantly surprised by the new options to fix transparancy on older content, which means classic London Underground routes no longer showed critical bugs such as huge spikes in the tunnels.

After entering all my settings and everything, I played the game, everything worked the same as before, **but there were two usability issues / changes which did not happen on my old install of v1.4**:

1. The cursor doesn't hide itself. When driving the train, whether or not you opened the menu since you loaded the route or not, the cursor does not hide at all, I've tried leaving it for 5 minutes too.

2. When alt tabbing, the resolution doesn't revert to your desktop resolution, if you use a different one. I use a 4:3 resolution as I often use the London Underground trains which have 2D cab images with a square aspect ratio. The experience is much, much more enjoyable.

On my old install of v1.4.xx, problem 1 would not occur as the cursor was never displayed in the first place. You never see the cursor in fullscreen mode, and in windowed mode, if you hover over the openBVE window the cursor hides itself.

On my old install of v1.4.xx, problem 2 didn't happen, instead if I alt-tabbed out of a fullscreen session of openBVE, my screen resolution would return to my default desktop one (it changed from my specified resolution of 1024x768 to 1920x1080) so I could continue using the desktop.

##### For problem 1:

In the options, et the window mode to Fullscreen, and load any train with any route. When you load in, your cursor will appear in the middle of the screen. It will not go away regardless of what you do (e.g. driving the train, invoking the menu, invoking then interacting with the menu, etc).

This issue keeps happening even if you try invoking the menu and then dismissing it again, interacting with the menu, etc.

This happens regardless if VSync is set to On or Off in the options, and regardless of the specified resolution in the windowed and fullscreen resolution fields (I tested)

##### For problem 2:

In the options, set the window mode to Fullscreen, and change the screen resolution for Fullscreen mode to something OTHER than your desktop resolution (like 1024x768 if you have a 1920x1080 monitor), then load any train with any route. Once loaded, ALT+TAB out of the game. The screen resolution stays at the resolution you specified earlier in options, and if you set it to a lower one, then it poses usability issues as the screen is restricted in size.

## Route

Not needed

## Train

Not needed

## Logs

Not needed (expected behaviour)

### Related information 

* Operating system: Windows 7 Ultimate 32-bit (SP1, v7601)
* Method of control: Keyboard ONLY

**Comment by sjain882** [comment]
Also a quick note: I can't import keyboard settings from v1.4 or v1.5, it just makes the whole Keyboard table blank, after throwing some errors.

And when I tried launching openBVE v1.7.1.0 after using v1.4.xx for some time, it throw loads of insane errors, then a unhandled windows form exception, presumably because it's trying to load some invalid option. You have to delete `%APPDATA%\openBVE` (that is Appdata > Roaming > openBVE) then only it will launch.

I backed up my v1.4.xx folder and switched the folders (after closing openBVE v1.7.xx), and launched v1.4.xx, yet the options for my v1.4.xx install still reset somehow, despite the Date Modified labels being intact and v1.7.xx not touching that v1.4.xx folder and its UserData\Settings folder.

If you'd like me to open a new issue for this and add logs, etc, let me know.

**Comment by leezer3** [comment]
Some thoughts:
* Importing keyboard settings from 1.4 is definitely not supported, sorry. (The windowing and input backend changed)
* Will check what's happening with the crash; IIRC the only issue should be the keyboard settings, which it *should* reset and popup an info window to this effect. Sounds like this isn't happening.
* ALT+TAB and the associated behaviour is controlled by the windowing backend, not by us directly. I'll look into this, but it may not be easily changable.
* Mouse cursor not hiding is semi-deliberate. We now support touch / click controls on the panel, and the menu obviously uses the mouse. Possibly ought to hide after inactivity, let me give it some thought :)

**Comment by leezer3** [comment]
So, I can't reproduce a crash with a settings folder generated by openBVE 1.4.3.0, which suggests there's something in yours it doesn't like.

Any chance of a copy of your bad settings folder zipped and attached here?
(Drag and drop into the comment box)

It may also have generated a crash log, which can be found by hitting the **Report Problem** link in the bottom left of the main form- This might tell me something useful :)
